source/app/controllers/urls_controller.rb

```diff
@@ -1,2 +1,57 @@
 class UrlsController < ApplicationController
+
+  #GET    /urls
+  def index
+    @urls = Url.all
+  end
+
+  #GET    /urls/new
+  def new
+    @url = Url.new
+  end
+  #GET    /urls/:id
+  def show
+    @url = Url.find(params[:id])
+  end
+
+  # POST   /urls
+  def create
+    puts "inside create"
+    @url = Url.new(url_params)
+    if @url.save
+      redirect_to url_path(@url)
+    else
+      render 'new'
+    end
+  end
+
+  #DELETE /urls/:id
+  def destroy
+    @url =Url.find(params[:id])
+    if @url.destroy
+      redirect_to urls_path
+    else
+
```

Comment:
`destroy` doesn't return true or false to tell you whether it succeeded, like some other ActiveRecord methods. So, just redirect unconditionally.

Reply:
Oh Thanks. Is it better if implement the error handling here? because, if destroy fails due to some DB issue it will raise an exception right?

Reply:
Great question. The `destroy` call can raise an error for many various reasons. I would not handle those errors in app code as long as we don't expect them from regular app usage. Make sure the app is connected with some error aggregation service, then let "exceptional" exceptions go uncaught.

If we start seeing errors we didn't account for, that's when we should do what we can to prevent them and also rescue them to return a more user-friendly response to our user.